visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts

```diff
@@ -26,23 +26,33 @@ export class FileChooserController {
 	public onImportNewFiles(element) {
 		this.$scope.$apply(() => {
 			this.storeService.dispatch(resetFiles())
+			let content
+
 			for (const file of element.files) {
+				const isCompressed = file.name.endsWith(".gz")
 				const reader = new FileReader()
 				reader.onloadstart = () => {
 					this.storeService.dispatch(setIsLoadingFile(true))
 				}
 				reader.onload = event => {
-					this.setNewData(file.name, (<any>event.target).result)
+					if (isCompressed) {
+						const pako = require("pako")
+						content = pako.ungzip((<any>event.target).result, { to: "string" })
```

Comment:
Please use the Node.js native unzip function https://nodejs.org/api/zlib.html. It is faster and does not require to use any extra library.